src/utilities/strategy-helpers/blue-green-helper.ts

```diff
@@ -0,0 +1,329 @@
+    const result = kubectl.apply(manifestFiles);
+
+    return { 'result': result, 'newFilePaths': manifestFiles };
+}
+
+export function getNewBlueGreenObject(inputObject: any, labelValue: string): object {
+    const newObject = JSON.parse(JSON.stringify(inputObject));
+
+    // Updating name only if label is green label is given
+    if (labelValue === GREEN_LABEL_VALUE) {
+        newObject.metadata.name = getBlueGreenResourceName(inputObject.metadata.name, GREEN_SUFFIX);
+    }
+
+    // Adding labels and annotations
+    addBlueGreenLabelsAndAnnotations(newObject, labelValue);
+
+    return newObject;
+}
+
+export function addBlueGreenLabelsAndAnnotations(inputObject: any, labelValue: string) {
+    //creating the k8s.deploy.color label
+    const newLabels = new Map<string, string>();
+    newLabels[BLUE_GREEN_VERSION_LABEL] = labelValue;
+
+    // updating object labels and selector labels
+    helper.updateObjectLabels(inputObject, newLabels, false);
+    helper.updateSelectorLabels(inputObject, newLabels, false);
+
+    // updating spec labels if it is a service
+    if (!helper.isServiceEntity(inputObject.kind)) {
+        helper.updateSpecLabels(inputObject, newLabels, false);
+    }
+}
+
+export function getBlueGreenResourceName(name: string, suffix: string) {
+    return `${name}${suffix}`;
+}
+
+export function getSpecLabel(inputObject: any): string {
+    if(!!inputObject && inputObject.spec && inputObject.spec.selector && inputObject.spec.selector.matchLabels && inputObject.spec.selector.matchLabels[BLUE_GREEN_VERSION_LABEL]) {
+        return inputObject.spec.selector.matchLabels[BLUE_GREEN_VERSION_LABEL]; 
+    }
+    return '';
+}
+
+export function getDeploymentMatchLabels(deploymentObject: any): string {
+    if (!!deploymentObject && deploymentObject.kind.toUpperCase()==KubernetesWorkload.pod.toUpperCase() &&  !!deploymentObject.metadata && !!deploymentObject.metadata.labels) {
+        return JSON.stringify(deploymentObject.metadata.labels);
+    } else if (!!deploymentObject && deploymentObject.spec && deploymentObject.spec.selector && deploymentObject.spec.selector.matchLabels) {
+        return JSON.stringify(deploymentObject.spec.selector.matchLabels);
+    }
+    return '';
+}
+
+export function getServiceSelector(serviceObject: any): string {
+    if (!!serviceObject && serviceObject.spec && serviceObject.spec.selector) {
+        return JSON.stringify(serviceObject.spec.selector);
+    } else return '';
+}
+
+export function isServiceSelectorSubsetOfMatchLabel(serviceSelector: string, matchLabels: string): boolean {
+    let serviceSelectorMap = new Map();
+    let matchLabelsMap = new Map();
+  
+    JSON.parse(serviceSelector, (key, value) => {
+      serviceSelectorMap.set(key, value);
+    });
+    JSON.parse(matchLabels, (key, value) => {
+      matchLabelsMap.set(key, value);
+    });
+  
+    let isMatch = true;
+    serviceSelectorMap.forEach((value, key) => {
```

Comment:
Instead use `every`?

Reply:
seems like `every `is not supported for maps